Backpack/Alert/Classes/BPKAlertView.m

```diff
@@ -0,0 +1,235 @@
+}
+
+- (void)setupConstraints {
+    [_circularView.topAnchor constraintEqualToAnchor:_circularView.superview.topAnchor].active = YES;
+    [_circularView.centerXAnchor constraintEqualToAnchor:_contentHolderView.centerXAnchor].active = YES;
+    [_circularView.heightAnchor constraintEqualToConstant:70].active = YES;
+    [_circularView.widthAnchor constraintEqualToConstant:70].active = YES;
+  
+    [_iconContainerView.centerYAnchor constraintEqualToAnchor:_circularView.centerYAnchor].active = YES;
+    [_iconContainerView.centerXAnchor constraintEqualToAnchor:_circularView.centerXAnchor].active = YES;
+    [_iconContainerView.heightAnchor constraintEqualToConstant:60].active = YES;
+    [_iconContainerView.widthAnchor constraintEqualToConstant:60].active = YES;
+   
+    [_iconImageView.centerYAnchor constraintEqualToAnchor:_iconContainerView.centerYAnchor].active = YES;
+    [_iconImageView.centerXAnchor constraintEqualToAnchor:_iconContainerView.centerXAnchor].active = YES;
+    
+    [_contentHolderView.topAnchor constraintEqualToAnchor:_circularView.bottomAnchor constant:-35].active = YES;
+    [_contentHolderView.bottomAnchor constraintEqualToAnchor:_contentHolderView.superview.bottomAnchor].active = YES;
+    [_contentHolderView.leadingAnchor constraintEqualToAnchor:_contentHolderView.superview.leadingAnchor].active = YES;
+    [_contentHolderView.trailingAnchor constraintEqualToAnchor:_contentHolderView.superview.trailingAnchor].active = YES;
+    
+    [_titleLabel.topAnchor constraintEqualToAnchor:_circularView.bottomAnchor constant:BPKSpacingLg].active = YES;
+    [_titleLabel.centerXAnchor constraintEqualToAnchor:_titleLabel.superview.centerXAnchor].active = YES;
+    
+    [_descriptionLabel.topAnchor constraintEqualToAnchor:_titleLabel.bottomAnchor constant:BPKSpacingLg].active = YES;
+    [_descriptionLabel.leadingAnchor constraintEqualToAnchor:_descriptionLabel.superview.leadingAnchor constant:BPKSpacingLg].active = YES;
+    [_descriptionLabel.trailingAnchor constraintEqualToAnchor:_descriptionLabel.superview.trailingAnchor constant:-BPKSpacingLg].active = YES;
+    
+    [_buttonStackView.topAnchor constraintEqualToAnchor:_descriptionLabel.bottomAnchor constant:BPKSpacingLg].active = YES;
+    [_buttonStackView.leadingAnchor constraintEqualToAnchor:_buttonStackView.superview.leadingAnchor constant:BPKSpacingXl].active = YES;
+    [_buttonStackView.trailingAnchor constraintEqualToAnchor:_buttonStackView.superview.trailingAnchor constant:-BPKSpacingXl].active = YES;
+    [_buttonStackView.bottomAnchor constraintEqualToAnchor:_buttonStackView.superview.bottomAnchor constant:-BPKSpacingLg].active = YES;
+}
+
+- (void)setupShadowOnView:(UIView *)view {
+    UIBezierPath *shadowPath = [UIBezierPath bezierPathWithRoundedRect:view.bounds cornerRadius:view.layer.cornerRadius];
+    view.layer.masksToBounds = NO;
+    view.layer.shadowColor = [UIColor colorWithRed:0.145 green:0.125 blue:0.200 alpha:1].CGColor;
```

Comment:
Any particular reason we aren't using `BPKShadow` here?

Reply:
Yes, let me explain:
- We have a circular view and a rectangle on top of each other and both should have a shadow. The problem is that its not 1 shape and putting shadows like BPKShadow on both of them will look bad as they are on top of each other. Instead of using CoreGraphics to do it in 1 shape I used this shortcut to have a BPK Shadow on the rectangle and a custom one on the circle that has an offset in the -y direction and 0 to the sides big enough that it would not show on the rectangle already but still looks the "same" and imitates like there is only 1 shadow on screen

Reply:
Can you use the suggestion @tevelee had? i.e compute a full shadow path and just have on shadow for that path?

Reply:
yeah give me a sec :)